docs/rules/strict.md

```diff
@@ -1,215 +1,228 @@
-    return;
+function foo2() {
+    // not strict mode
 };
+
+(function() {
+    "use strict";
+    function bar() {
+        // strict mode
+    }
+}());
 ```
 
-Unlike scripts, ECMAScript modules are always in strict mode. Strict mode directives in ECMAScript modules have no effect.
+In Node.js and the CommonJS module system, a hidden function wraps each module and limits the scope of a "global" strict mode directive.
+
+In ECMAScript modules, which always have strict mode semantics, the directives are unnecessary.
 
 ## Rule Details
 
-This rule is aimed at using strict mode directives effectively, and as such, will flag any unexpected uses or omissions of strict mode directives.
+This rule requires or disallows strict mode directives.
 
-## Options
+This rule reports all strict mode directives as problems if ECMAScript modules or implied strict mode are enabled (see [Specifying Parser Options](../user-guide/configuring#specifying-parser-options)). This behaviour does not depend on the rule options, but can be silenced by disabling this rule.
 
-There are four options for this rule:
+## Options
 
-* `"safe"` - require `"use strict"` globally when inside a module wrapper and in function scopes everywhere else. This is the default.
-* `"never"` - disallow `"use strict"`.
-* `"global"` - require `"use strict"` in the global scope.
-* `"function"` - require `"use strict"` in function scopes only.
+This rule has a string option:
 
-All strict mode directives are flagged as unnecessary if ECMAScript modules or implied strict mode are enabled (see [Specifying Parser Options](../user-guide/configuring#specifying-parser-options)). This behaviour does not depend on the rule options, but can be silenced by disabling this rule.
+* `"safe"` (default) corresponds either of the following options:
+    * `"global"` if code is in a module wrapper
+    * `"function"` if code is not in a module wrapper
+* `"global"` requires one strict mode directive in the global scope (and disallows any other strict mode directives)
+* `"function"` requires one strict mode directive in each top-level function declaration or expression (and disallows any other strict mode directives)
+* `"never"` disallows strict mode directives
 
 ### safe
 
-Node.js and the CommonJS module system wrap modules inside a hidden function wrapper that defines each module's scope. The wrapper makes it safe to concatenate strict mode modules while maintaining their original strict mode directives. When the `node` or `commonjs` environments are enabled or `globalReturn` is enabled in `ecmaFeatures`, ESLint considers code to be inside the module wrapper, and `"safe"` mode corresponds to `"global"` mode and enforces global strict mode directives. Everywhere else, `"safe"` mode corresponds to `"function"` mode and enforces strict mode directives inside top-level functions.
+The `"safe"` option corresponds to the `"global"` option if ESLint considers a file to consist of code in a module wrapper:
```

Comment:
The `"safe"` option corresponds to the `"global"` option if ESLint considers a file to be a **Node.js** or **CommonJS** module because the configuration specifies either of the following

Thank you for sharing expert understanding. Made changes up to and including **safe** section.